File: core/network/i_network.go
package interfaces

type GetParams struct {
	URL             string
	QueryParameters map[string]string
	Headers         map[string]string
}

type PostParams struct {
	URL             string
	QueryParameters map[string]string
	Headers         map[string]string
	Body            interface{}
}

type PutParams struct {
	URL             string
	QueryParameters map[string]string
	Headers         map[string]string
	Body            interface{}
}

type DeleteParams struct {
	URL             string
	QueryParameters map[string]string
	Headers         map[string]string
}

type PatchParams struct {
	URL             string
	QueryParameters map[string]string
	Headers         map[string]string
	Body            interface{}
}

type INetwork interface {
	Get(params GetParams) ([]byte, error)
	Post(params PostParams) ([]byte, error)
	Put(params PutParams) ([]byte, error)
	Delete(params DeleteParams) ([]byte, error)
	Patch(params PatchParams) ([]byte, error)
}

File: frontend/pokechancepocket/src/core/network/inetwork.ts
export interface getParams {
  url: string;
  header?: Record<string, string>;
}

export default interface INetwork {
  get<T>(params: getParams): Promise<T>;
  post<T>(url: string, data: any): Promise<T>;
  put<T>(url: string, data: any): Promise<T>;
  delete<T>(url: string): Promise<T>;
}
